### PR TITLE
Refactor ActionManagementDAOImpl to have NamedJdbcTemplate template

### PR DESCRIPTION
**Pull request by ashanthamara**
### Proposed changes in this pull request

With this improvement it can be easily handle the database transactions when the ActionManagementDAOImpl calls the SecretManagement service to encrypt/decrypt secrets.

### Related Issue
- https://github.com/wso2/product-is/issues/21594

### CODE REVIEW

File: components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/constant/ActionMgtConstants.java [hunk 1]
```diff
@@ -86,7 +86,9 @@ public enum ErrorMessages {
                 "Error while decrypting Action Endpoint Authentication properties in the system."),
         ERROR_NO_AUTHENTICATION_TYPE("65013",
                 "Error while retrieving Action Endpoint Authentication configurations",
-                "Authentication type is not defined for the Action Endpoint.");
+                "Authentication type is not defined for the Action Endpoint."),
+        ERROR_WHILE_UPDATING_ACTION_BASIC_INFO("65014", "Error while updating Action basic info",
```
Comment (malithie) [inline components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/constant/ActionMgtConstants.java:90]: Shall we term this as "Error while updating basic Action information"
Fix the description accordingly

File: components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java [hunk 2]
```diff
@@ -63,145 +61,134 @@ public ActionManagementDAOImpl() {
...
-                    LOG.debug(String.format("Error while creating the Action of Action Type: %s in Tenant Domain: %s." +
-                                    " Rolling back created action information and deleting added secrets.", actionType,
-                            IdentityTenantUtil.getTenantDomain(tenantId)));
-                }
-                actionSecretProcessor.deleteAssociatedSecrets(action.getEndpoint().getAuthentication(), actionId);
-                IdentityDatabaseUtil.rollbackTransaction(dbConnection);
-                throw ActionManagementUtil.handleServerException(
-                        ActionMgtConstants.ErrorMessages.ERROR_WHILE_ADDING_ACTION, e);
+
+                return null;
```
Comment (malithie) [inline components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java:86]: Is a return necessary here ?

File: components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java [hunk 3]
```diff
@@ -63,145 +61,134 @@ public ActionManagementDAOImpl() {
...
+                    return null;
+                }, statement -> {
+                    statement.setString(ActionMgtSQLConstants.Column.ACTION_TYPE, actionType);
+                    statement.setInt(ActionMgtSQLConstants.Column.TENANT_ID, tenantId);
+                });
+
+            for (Map.Entry<String, Action.ActionResponseBuilder> entry : actionBuilderMap.entrySet()) {
+                EndpointConfig endpoint = getActionEndpointConfigById(entry.getKey(), tenantId);
+                actions.add(entry.getValue().endpoint(endpoint).build());
             }
```
Comment (malithie) [inline components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java:130]: Looks like this refactoring has introduced another additional loop to go over all actions and to get the endpoint details.

File: components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/constant/ActionMgtConstants.java (hunk 1 above)
Comment (ashanthamara) [reply components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/constant/ActionMgtConstants.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/6097/commits/d824f6d306b493c78505a4d096ff497128b6ffe3

File: components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java [hunk 4]
```diff
@@ -391,115 +372,97 @@ private void addEndpointProperties(Connection dbConnection, String actionId,
...
-        } catch (SQLException e) {
-            throw ActionManagementUtil.handleServerException(
-                    ActionMgtConstants.ErrorMessages.ERROR_WHILE_RETRIEVING_ACTION_ENDPOINT_PROPERTIES, e);
+
+            return new EndpointConfig.EndpointConfigBuilder()
+                    .uri(actionEndpointProperties.get(ActionMgtConstants.URI_ATTRIBUTE))
+                    .authentication(authentication).build();
+        } catch (ActionMgtServerException | DataAccessException e) {
+            throw new SQLException(
```
Comment (malithie) [inline components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java:464]: Better add a comment on the reason to throw the SQL exception as this exception is really not of SQL type

Comment (malithie) [reply components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java:464]: Or we need to have a runtime exception and handle that specifically

Comment (ashanthamara) [reply components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java]: Added a new Runtime exception with https://github.com/wso2/carbon-identity-framework/pull/6097/commits/d89041e658585942ac7478a30c6e57b5c503f09a

**Review by malithie — APPROVED** [review]

**Pull request merged**